streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -17,242 +17,225 @@
-        log.debug("Deleting topic {} from ZK in partition assignor.", topic);
+        String brokerId = Integer.toString(brokerNode.id());
+        RequestSend send = new RequestSend(brokerId,
+                kafkaClient.nextRequestHeader(ApiKeys.CREATE_TOPICS),
+                createTopicsRequest.toStruct());
 
-        zkClient.createPersistent(ZK_DELETE_TOPIC_PATH + "/" + topic, "", ZooDefs.Ids.OPEN_ACL_UNSAFE);
+        sendRequest(send, brokerNode);
     }
 
-    private void addPartitions(String topic, int numPartitions, int replicationFactor, Map<Integer, List<Integer>> existingAssignment) {
-        log.debug("Adding {} partitions topic {} from ZK with existing partitions assigned as {} in partition assignor.", topic, numPartitions, existingAssignment);
 
-        List<Integer> brokers = getBrokers();
-        int numBrokers = brokers.size();
-        if (numBrokers < replicationFactor) {
-            log.warn("Not enough brokers found. The replication factor is reduced from " + replicationFactor + " to " +  numBrokers);
-            replicationFactor = numBrokers;
-        }
+    /**
+     * Delere an existing topic
+     *
+     * @param topic
+     * @param brokerNode
+     */
+    private void deleteTopic(String topic, Node brokerNode) {
+        log.debug("Deleting topic {} from ZK in partition assignor.", topic);
+        Set<String> topics = new HashSet();
+        topics.add(topic);
 
-        int startIndex = existingAssignment.size();
+        DeleteTopicsRequest deleteTopicsRequest = new DeleteTopicsRequest(topics, config.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
 
-        Map<Integer, List<Integer>> newAssignment = new HashMap<>(existingAssignment);
+        String brokerId = Integer.toString(brokerNode.id());
+        RequestSend send = new RequestSend(brokerId,
+                kafkaClient.nextRequestHeader(ApiKeys.DELETE_TOPICS),
+                deleteTopicsRequest.toStruct());
 
-        for (int i = 0; i < numPartitions; i++) {
-            ArrayList<Integer> brokerList = new ArrayList<>();
-            for (int r = 0; r < replicationFactor; r++) {
-                int shift = r * numBrokers / replicationFactor;
-                brokerList.add(brokers.get((i + shift) % numBrokers));
+        sendRequest(send, brokerNode);
+    }
+
+    /**
+     * Send a requet to a node.
+     *
+     * @param send
+     * @param brokerNode
+     */
+    private void  sendRequest(RequestSend send, Node brokerNode) {
+        RequestCompletionHandler callback = new RequestCompletionHandler() {
+            public void onComplete(ClientResponse response) {
+                handleResponse(response);
+            }
+        };
+
+        ClientRequest clientRequest = new ClientRequest(new SystemTime().milliseconds(), true, send, callback);
+        SystemTime systemTime = new SystemTime();
+        int iterationCount = 0;
+        while (iterationCount < 5) { // TODO: Set this later
```

Comment:
Same as above comments on the loop. `iterationCount` is never incremented and i think the loop could be simplified a bit. Also, as it is the same logic it is probably worth extracting it into a method, i.e, `isClientReady` or sth and removing the duplication